mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/dto/DeletedMessageWithStorageInformationDTO.java

```diff
@@ -0,0 +1,272 @@
+                deletedMessage.getSender().asString(),
+                getRecipients(deletedMessage.getRecipients()),
+                deletedMessage.getSubject(),
+                deletedMessage.hasAttachment(),
+                deletedMessage.getSize());
+        }
+
+        private final String messageId;
+        private final List<String> originMailboxes;
+        private final String owner;
+        private final String deliveryDate;
+        private final String deletionDate;
+        private final String sender;
+        private final List<String> recipients;
+        private final Optional<String> subject;
+        private final boolean hasAttachment;
+        private final long size;
+
+        @JsonCreator
+        public DeletedMessageDTO(@JsonProperty("messageId") String messageId,
+                                 @JsonProperty("originMailboxes") List<String> originMailboxes,
+                                 @JsonProperty("owner") String owner,
+                                 @JsonProperty("deliveryDate") String deliveryDate,
+                                 @JsonProperty("deletionDate") String deletionDate,
+                                 @JsonProperty("sender") String sender,
+                                 @JsonProperty("recipients") List<String> recipients,
+                                 @JsonProperty("subject") Optional<String> subject,
+                                 @JsonProperty("hasAttachment") boolean hasAttachment,
+                                 @JsonProperty("size") long size) {
+            this.messageId = messageId;
+            this.originMailboxes = originMailboxes;
+            this.owner = owner;
+            this.deliveryDate = deliveryDate;
+            this.deletionDate = deletionDate;
+            this.sender = sender;
+            this.recipients = recipients;
+            this.subject = subject;
+            this.hasAttachment = hasAttachment;
+            this.size = size;
+        }
+
+        public String getMessageId() {
+            return messageId;
+        }
+
+        public List<String> getOriginMailboxes() {
+            return originMailboxes;
+        }
+
+        public String getOwner() {
+            return owner;
+        }
+
+        public String getDeliveryDate() {
+            return deliveryDate;
+        }
+
+        public String getDeletionDate() {
+            return deletionDate;
+        }
+
+        public String getSender() {
+            return sender;
+        }
+
+        public List<String> getRecipients() {
+            return recipients;
+        }
+
+        public Optional<String> getSubject() {
+            return subject;
+        }
+
+        public boolean getHasAttachment() {
+            return hasAttachment;
+        }
+
+        public long getSize() {
+            return size;
+        }
+
+        private static ImmutableList<String> getOriginMailboxes(List<MailboxId> originMailboxes) {
```

Comment:
static goes first

Reply:
serializeOriginMailboxes

Reply:
I don't think static goes before the private..?

Reply:
static fields

public static

private static

fields

constructor

public methods

private methods

Reply:
Misunderstood, sorry